### PR TITLE
PR announcement spam prevention

### LINKED ISSUE #14508: The PR announcement system is very vulnerable to being spammed
Since github bans are apparently awful to apply, we were at the mercy of someone just spam opening/closing his PR until we kicked the bots and disabled the announcements serverside. Needs flood detection or something.

@MrStonedOne

### PR DESCRIPTION
**Pull request by Cyberboss**
Limits unique PR messages to 3 per round. So that's "Opened, Reopened, Closed" * Author | Maintainer

Allows the maintainers to be seen without the author being able to spam

Fixes #14508

### CODE REVIEW

File: code/world.dm [hunk 1]
```diff
@@ -187,6 +183,24 @@ var/last_irc_status = 0
 		else
 			return ircadminwho()
 
+#define PR_ANNOUNCEMENTS_PER_ROUND 3 //The nunber of unique PR announcements allowed per round
```
Comment (lzimann) [inline code/world.dm]: nuNber

**Comment by WJohn** [comment]
Goofball is why we can't have nice things.

**Comment by optimumtact** [comment]
Technically I am the reason for this

**Comment by Iamgoofball** [comment]
:-1:

This just means I can abuse this to slip stuff past players.

**Comment by WJohn** [comment]
Outplayed by goof once again.

**Comment by PKPenguin321** [comment]
👎 because of what goof said, and if one guy makes a PR then opens and closes it over and over forever we can just ban him from the game/github

although i could see a bot being made to automate the spam and evade bans, in which case this would be needed, yeah

**Comment by Iamgoofball** [comment]
# we have a preference to hide PR announcements

**Comment by Cyberboss** [comment]
PK this won't hide shit. The way it works is a round will announce the
first 3 opens and closes FOR A SPECIFIC PR/USER COMBINATION and then ignore
the rest for that round

**Comment by MrStonedOne** [comment]
Ya, but i'd rather users don't all start hiding them because of some asshat.

anywho, cyberboss, this has a weakness, you can just change the title to get around it.

I'll change the announcer to send the entire pull request JSON object over and you can key by pr id instead.

Then have it be 3 actions (of any kind) and just exempt when somebody other then the pr creator is the one doing the action.

#24008

**Comment by Iamgoofball** [comment]
@MrStonedOne 
You can mute people from the announcer already, why not just use that on people who abuse it?

**Comment by JamieH** [comment]
We've had this issue - had to totally disable the announcer. They were either renaming there GitHub account or deleting it too, so we couldn't block them from our repo in time.

**Comment by optimumtact** [comment]
Conflicts

**Comment by MrStonedOne** [comment]
this is also still waiting for the bugs i mentioned to be fixed by taking advance of #24008

**Comment by Cyberboss** [comment]
100% laziness

**Comment by optimumtact** [comment]
Ping for reopen or make new pr when you're 100% less lazy

**Comment by Cyberboss** [comment]
Vacation

**Comment by MrStonedOne** [comment]
@Cyberboss you never resumed work on this.

**Pull request merged**